typeagent/aitools/embeddings.py

```diff
@@ -188,18 +209,81 @@ def hashish(s: str) -> int:
             result = np.array(fake_data, dtype=np.float32)
             return result
         else:
-            # TODO: Split in batches of 2048 inputs if too long;
-            # or smaller if inputs are large.
-            data = (
-                await self.async_client.embeddings.create(
-                    input=input,
-                    model=self.model_name,
-                    encoding_format="float",
-                    **extra_args,
-                )
-            ).data
-            assert len(data) == len(input), (len(data), "!=", len(input))
-            return np.array([d.embedding for d in data], dtype=np.float32)
+
+            chunked_groups = {}
```

Comment:
Here starts a very long block of code that I think would gain a lot of clarity if you refactored it into a few helper functions/methods.